src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -707,6 +707,16 @@ def exec_line_for_queue(executor_creation_func, input_queue: Queue, output_queue
         try:
             data = input_queue.get(timeout=1)
             if data == TERMINATE_SIGNAL:
+                try:
+                    import opentelemetry.trace as otel_trace
+
+                    # Meet span missing issue when end process normally.
+                    # Shutdown the span processor to flush the remaining spans.
+                    # The span processor is created for each process, so it's ok to shutdown it here.
+                    otel_trace.get_tracer_provider()._active_span_processor.shutdown()
```

Comment:
There might be multiple span processors. How do we handle that case?

Reply:
`_active_span_processor` is a list. Shutdown() will shut down them all.